packages/rum/src/domain/profiling/profiler.ts

```diff
@@ -31,15 +43,19 @@ export function createRumProfiler(
 ): RUMProfiler {
   const isLongAnimationFrameEnabled = supportPerformanceTimingEvent(RumPerformanceEntryType.LONG_ANIMATION_FRAME)
 
+  let lastViewEntry: RumViewEntry | undefined
+
   let instance: RumProfilerInstance = { state: 'stopped' }
 
-  function start(viewId: string | undefined): void {
+  function start(viewEntry: ViewHistoryEntry | undefined): void {
     if (instance.state === 'running') {
       return
     }
 
     // Add initial view
-    collectViewEntry(viewId)
+    lastViewEntry = viewEntry
+      ? { startTime: performance.now(), viewId: viewEntry.id, viewName: viewEntry.name }
```

Comment:
The view name is computed in the backend. It's only available in the frontend when the customer wants to override it using `startView` API.

Reply:
What do you suggest we use ? In practice it seems to be defined and sensical, cf. [link](https://dd.datad0g.com/profiling/explorer?query=service%3Aweb-ui&agg_m=%40prof_chrome_cpu_cores&agg_m_source=base&agg_q=service&agg_q_source=base&agg_t=sum&event=AwAAAZX14CS3A0TO-AAAABhBWlgxNENVRkFBQ3dOcmlfajNHUXVBQUEAAAA1MDE5NWY1ZTMtYzU1ZS00MTYyLWJhZTctOWQ3ZTczZDlmMDZlLXN5bnRoZXRpYy1saXN0LTAAAuSA&fromUser=false&my_code=disabled&profileId=AZX14CUFAACwNri_j3GQuAAA&top_n=100&top_o=top&viz=stream&x_missing=true&start=1743572462631&end=1743586862631&paused=true)

Reply:
also for the startTime you should probably do:
```suggestion
      ? { startTime: viewEntry.startClock.startClocks.relative, viewId: viewEntry.id, viewName: viewEntry.name }
```

Reply:
Oh so re-reading this, you mean the `viewName` is actually set for us because we are using the `startView` API, but could be empty for customers not using that API ? 
I believe that's fine. We are using the `viewName` to better categorize the samples we get from the Profiler, and thus creating filters per "views" like the following: 
![Screenshot 2025-04-02 at 17 05 48](https://github.com/user-attachments/assets/7d1e044d-e6be-415a-9e3c-5157bdd4f714)

I'm happy to change this, but at the same time I'm fine dealing with empty values in our processing, and use it whenever it's available as I find it to be valuable.

Reply:
Yes at Datadog we are using `startView`, this is why it works for us. However `startView` API is used by less than 10% of customers, so if categorizing by viewName is valuable, we should consider a different strategy. I'm fine merging this for internal dogfooding, but we should put more thought into it before GA.

Reply:
Updated code to use `viewEntry.startClocks.relative` instead of `performance.now()`. I confirmed it was similar (except for initial view but that's even better)

Also, as agreed together, let's keep usage of `view.name` for dogfooding and start discussion around better support for more generic view name handling in Profiling.